test/github-issues/660/issue-660.ts

```diff
@@ -0,0 +1,116 @@
+
+        if (connection.driver instanceof SqlServerDriver) {
+            expect(sql).to.equal("INSERT INTO user(name) OUTPUT INSERTED.* VALUES (@0)"); }
+        else if (connection.driver instanceof PostgresDriver) {
+            expect(sql).to.equal("INSERT INTO user(name) VALUES ($1) RETURNING INSERTED.*"); }
+        // else { // this is arguably an error case, since .returning() is only supported by PostgreSQL and MSSQL
+        //     expect(sql).to.equal("INSERT INTO user(name) VALUES (@0)"); }
+    })));
+
+    it("should create an UPDATE statement, including RETURNING or OUTPUT clause as appropriate", () => Promise.all(connections.map(async connection => {
+
+        const sql = connection.createQueryBuilder()
+            .update(User)
+            .set({ name: "Joe Bloggs" })
+            .where("name = :name", { name: user.name })
+            .returning("INSERTED.*")
+            .disableEscaping()
+            .getSql();
+
+        if (connection.driver instanceof SqlServerDriver) {
+            expect(sql).to.equal("UPDATE user SET name = @0 OUTPUT INSERTED.* WHERE name = @1"); }
+        else if (connection.driver instanceof PostgresDriver) {
+            expect(sql).to.equal("UPDATE user SET name = $1 RETURNING INSERTED.* WHERE name = $2"); } // Fudged, until PostgreSQL RETURNING clause can be correctly placed after WHERE clause
+        // else { // this is arguably an error case, since .returning() is only supported by PostgreSQL and MSSQL
+        //     expect(sql).to.equal("UPDATE user SET name = @0 WHERE name = @1"); }
+    })));
+
+    it("should perform update with OUTPUT correctly", () => Promise.all(connections.map(async connection => {
+
+        await connection.manager.save(user);
+
+        if (connection.driver instanceof SqlServerDriver) {
+            // should also be: || connection.driver instanceof PostgresDriver) but only when PostgreSQL RETURNING clause can be correctly placed after WHERE clause
+            const updateUser = await connection.createQueryBuilder()
+                .update(User)
+                .set({ name: "Joe Bloggs" })
+                .where("name = :name", { name: user.name })
+                .returning("INSERTED.*")
+                .execute();
+    
+            updateUser.should.be.eql([
+                { id: 1, name: "Joe Bloggs" }
+            ]);
+        }
+    })));
+
+    it("should create a DELETE statement, including RETURNING or OUTPUT clause as appropriate", () => Promise.all(connections.map(async connection => {
+
+        // await connection.manager.save(user);
+
+        const sql = connection.createQueryBuilder()
+            .delete()
+            .from(User)
+            .where("name = :name", { name: user.name })
+            .returning("DELETED.*")
+            .disableEscaping()
+            .getSql();
+
+        if (connection.driver instanceof SqlServerDriver) {
+            expect(sql).to.equal("DELETE FROM user OUTPUT DELETED.* WHERE name = @0"); }
```

Comment:
currently CI is not configured to run sql server tests... You are able to run them manually?